puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-x-develop-release.groovy

```diff
@@ -10,37 +10,46 @@ pipeline {
         ansiColor('xterm')
     }
 
-    environment {
-        ruby_version = "2.5"
-    }
-
     stages {
-        stage('Copy Source') {
-            steps {
-                script {
-                    sourcefile_paths = pwd(tmp: true)
-                    source_project_name = "${project_name}-${git_ref}-source-release"
-                    copyArtifacts(projectName: source_project_name, target: sourcefile_paths, selector: {lastSuccessful: true})
-                }
-            }
-        }
         stage('Build Package') {
             parallel {
                 stage('Build RPM') {
                     when {
                         expression { build_rpm }
                     }
-                    steps {
-                        dir('foreman-packaging') {
-                            obal(
-                                action: 'nightly',
-                                packages: obal_package_name,
-                                extraVars: [
-                                    'releasers': releasers,
-                                    'nightly_sourcefiles': sourcefile_paths,
-                                    'nightly_githash': commit_hash
-                                ]
-                            )
+                    stages {
+                        stage('Copy Source') {
+                            steps {
+                                script {
+                                    artifact_path = "${pwd()}/artifacts"
+                                    source_project_name = "${project_name}-${git_ref}-source-release"
+                                    copyArtifacts(projectName: source_project_name, target: artifact_path)
+                                    sourcefile_paths = list_files("${artifact_path}/*").findAll { it.endsWith('.gem') || it.endsWith('tar.bz2') || it.endsWith('tar.gz') }
+                                }
+                            }
+                        }
+                        stage('Setup Environment') {
+                            steps {
+                                dir('foreman-packaging') {
+                                    git(url: 'https://github.com/theforeman/foreman-packaging.git', branch: 'rpm/develop', poll: false)
+                                }
+                                setup_obal()
+                            }
+                        }
+                        stage('Release') {
+                            steps {
+                                dir('foreman-packaging') {
+                                    obal(
+                                        action: 'nightly',
+                                        packages: obal_package_name,
+                                        extraVars: [
+                                            'releasers': releasers,
+                                            'nightly_sourcefiles': sourcefile_paths,
+                                            'nightly_githash': commit_hash
```

Comment:
Looks like in cfe5cc6eaf64e927280d13ab8ded56e081c212a1 I dropped this and broke it. I think we need to read this from the `commit` artifact.
```suggestion
                                            'nightly_githash': readFileFromWorkspace("${artifact_path}/commit")
```

(and remove `commit_hash` from the first line)

Reply:
Reading it from the copied artifacts now, check it out

Reply:
https://ci.theforeman.org/job/hammer-cli-katello-master-package-release/67/artifact/extra_vars-1582563779921-4caaa1ad-462e-4206-9046-0cbe55dc0455.yaml/*view*/